### ISSUE TITLE
request to add layer "preadd" event 

### ISSUE THREAD

**Issue by rendrom** [Leaflet/Leaflet#5686]
Problem:
https://jsfiddle.net/rendrom/n4c6v6fs/

Impossible to catch tile loading events to the newly added layer

Possible fix:
https://jsfiddle.net/rendrom/0bef9r7z/

It is possible to add extra map 'prelayeradd' and L.Layer 'preadd' events at the beginning of the L.Layer.prototype._layerAdd method (like in  fix example)?

Another way to fix this is to move up fire action inside _layerAdd method.

**Comment by perliedman** [comment]
Hi,
I understand your problem as: the `layeradd` event is fired _after_ the layer has actually been added, and the layer's `onAdd` method has executed. In your particular case, this is a problem, since a `GridLayer` (and `TileLayer`) will already have fired its `tileload` events at this point, so adding a listener is "too late".

Is this a correct description of the problem?

**Comment by rendrom** [comment]
@perliedman, exactly!
Long ago I ran into this problem. I making changes for myself in each Leaflet version. But now decided to report.

When using layers with a long tile load, it is inconvenient that you can not display the loading indicator immediately after adding this layer to the map

**Comment by FadilEldin** [comment]
The problem code: https://jsfiddle.net/rendrom/n4c6v6fs/ and the fix code https://jsfiddle.net/rendrom/0bef9r7z/ provide the same user experience to me. or am I wrong?

**Comment by rendrom** [comment]
Here are updates for the Leaflet 1.6

problem: https://jsfiddle.net/rendrom/p7tz3xro/
fix: https://jsfiddle.net/rendrom/w1ernc2z/

**Comment by FadilEldin** [comment]
Yes, with the new examples, the load indicator doe not disappear, and this seems to solve the issue. 
How this solution can be merged into leaflet?

**Comment by rendrom** [comment]
I think, before merged, it is necessary to discuss and decide is it approach suitable for leaflet api.

**Comment by Falke-Design** [comment]
Is this a duplicate of #2967?

**Comment by johnd0e** [comment]
Initial problem is the same as in 2967, but it is still unclear if we need 'preadd' or not.
